src/middlewares/auth-middleware.ts

```diff
@@ -48,104 +47,24 @@ async function authentication(
-            parentError: error,
-          });
-        });
-        span.setDisclosedAttribute("database reachable", true);
-
-        if (!user) {
-          span.setDisclosedAttribute("user found on DB", false);
-          throw new NoDBUserFoundError();
+        if (request.url && request.url.includes("/api/")) {
+          throw webErrorFactory(webErrors.unauthorized);
         }
-        span.setDisclosedAttribute("user found on DB", true);
-
-        const refreshedTokens = await oauth2Client
-          .refreshTokens(user.refresh_token)
-          .catch((error) => {
-            span.setDisclosedAttribute("is token refreshed", false);
-            if (error instanceof UnreachableError) {
-              throw webErrorFactory({
-                ...webErrors.unreachable,
-                parentError: error,
-              });
-            }
-
-            throw error;
-          });
-        span.setDisclosedAttribute("is token refreshed", true);
-
-        await decryptVerifyAccessToken(refreshedTokens.access_token).catch(
-          (error) => {
-            span.setDisclosedAttribute("is access_token valid", false);
-            if (
-              error instanceof MissingJWKSURIError ||
-              error instanceof InvalidAudienceError ||
-              error instanceof MissingKeyIDHS256Error ||
-              error instanceof AlgoNotSupportedError ||
-              error instanceof InvalidKeyIDRS256Error ||
-              error instanceof UnhandledTokenType
-            ) {
-              throw webErrorFactory({
-                ...webErrors.badRequest,
-                parentError: error,
-              });
-            }
-
-            if (error instanceof UnreachableError) {
-              throw webErrorFactory({
-                ...webErrors.unreachable,
-                parentError: error,
-              });
-            }
-
-            throw error;
-          },
-        );
-        span.setDisclosedAttribute("is refresh access_token valid", true);
-
-        await setServerSideCookies(
-          response,
-          "user-cookie",
-          {
-            access_token: refreshedTokens.access_token,
-            sub,
-          },
-          {
-            shouldCookieBeSealed: true,
-            cookieSalt: configVariables.cookieSalt,
-            maxAge: 24 * 60 * 60,
-            path: "/",
-            httpOnly: true,
-            secure: true,
-          },
-        );
 
-        span.setDisclosedAttribute("is cookie set", true);
-
-        await getAndPutUser(
-          { sub, refresh_token: refreshedTokens.refresh_token },
-          user,
-        ).catch((error) => {
-          span.setDisclosedAttribute("database reachable", false);
-          throw webErrorFactory({
-            ...webErrors.databaseUnreachable,
-            parentError: error,
-          });
-        });
-
-        span.setDisclosedAttribute("user updated on DB", true);
+        const redirectURL = !request.url ? "/account" : request.url;
```

Comment:
🙃 Isn't simpler to:?
```suggestion
        const redirectURL = request.url ? request.url : "/account";
```

Reply:
"Why make it simple when you can make it complicated" 😅   
Joke aside, your suggestion is far better and easier to read.